### ISSUE TITLE
ESLint fails when un-ignoring a file in an ignored directory in Windows

### ISSUE THREAD

**Issue by ratherblue** [eslint/eslint#4408]
I made a demo repository here that reproduces the issue: https://github.com/ratherblue/eslint-ignore-bug.

ESLint version: 1.9.0 (happens at 1.4.0 and up)
Windows version: Windows 7

When a directory is ignored in `.eslintignore`, and then a file underneath it is *unignored*, the linter fails. This does not happen in OS X.

Steps to reproduce on Windows:

1. `git clone https://github.com/ratherblue/eslint-ignore-bug.git`
2. `cd eslint-ignore-bug`
3. `npm install`
4. `node_modules\.bin\eslint **\*.js`

Expected: Should report `js/xyz.js` as having lint errors

Actual: No errors are reported

If you edit `.eslintignore` and remove `!js/sub/123.js`, then `js/xyz.js` is correctly reported as having errors.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Closing as a duplicate of the #4136 We do not currently support `!` pattern in eslintignore. Feel free to reopen if you think this is different then the issue above.

**Comment by ratherblue** [comment]
This isn't the same as #4136. This is documented as a feature of ESLint. If you no longer support it you should update the information on this page: http://eslint.org/docs/user-guide/configuring

> * Lines preceded by ! are negated patterns that re-include a pattern that was ignored by an earlier pattern.

> ````
> # Ignore built files except build/index.js
> build/
> !build/index.js
> ````

It still works in OS X and it used to work in Windows until version 1.4.0, so it would be a great feature to have back in so that my configuration can be consistent between operating systems.

**Comment by ratherblue** [comment]
@ilyavolodin Can you please reopen this bug?

**Comment by nzakas** [comment]
We are completely rewriting how ignores work for 2.0.0. There are already plenty of issues tracking this, and we won't be making any other changes prior to that.
